src/lib/storage/table.cpp

```diff
@@ -151,13 +151,27 @@ void Table::append_mutable_chunk() {
 }
 
 uint64_t Table::row_count() const {
-  uint64_t ret = 0;
+  if (_type == TableType::References && _cached_row_count && !HYRISE_DEBUG) {
+    return *_cached_row_count;
+  }
+
+  uint64_t row_count = 0;
   const auto chunk_count = _chunks.size();
   for (auto chunk_id = ChunkID{0}; chunk_id < chunk_count; ++chunk_id) {
     const auto chunk = get_chunk(chunk_id);
-    if (chunk) ret += chunk->size();
+    if (chunk) row_count += chunk->size();
```

Comment:
Where do we cover the case of overallocated mutable chunks? At first sight, I did not find it in chunk.cpp or value_segment.cpp (chunk.cpp uses the first segment's size).

Reply:
This is not part of `size()` or `row_count()` anywhere. The preallocation happens solely in the `capacity()` of the vectors.

Reply:
What a dump question that was. 🤦🏻‍♂️